src/hooks/useBalance/useEVMBalance.tsx

```diff
@@ -1,30 +1,46 @@
-import { useCallback, useEffect, useState } from 'react'
-
+import { useCallback, useEffect } from 'react'
+import type { Web3Provider } from '@ethersproject/providers'
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+import { useTabActive } from '../useTabActive/useTabActive'
 import type { TUseBalanceOptions } from './types'
 import { isEvmWallet } from '@/utils/wallet'
 
+const balanceFetcher = (options: TUseBalanceOptions) => {
+  const { isConnected, provider, address } = options
+  const isReadyForRequest = isEvmWallet(options) && address && isConnected && provider
+
+  if (!isReadyForRequest) {
+    return
+  }
+
+  return (provider as Web3Provider).getBalance(address).then(String)
+}
+
 function useEVMBalance(options: TUseBalanceOptions) {
-  const { isConnected, provider, name, address, chainId } = options
+  const { isConnected, provider, address, chainId, updateDelay } = options
   const isSubscriptionIsAvailable = isEvmWallet(options) && address && isConnected && provider
-
-  const [balance, setBalance] = useState<string | null>(null)
+  const isTabActive = useTabActive()
+  const queryClient = useQueryClient()
+  const { data: balance } = useQuery(['evmBalance', address], () => balanceFetcher(options), {
+    initialData: null,
```

Comment:
❓The `initialData` parameter tells the hook that we have the initial data. Although we don't, by default we don't know the balance. I don't understand why it's done this way? To avoid the first request?